### PR TITLE
Invalid return type for Price Currency Interface `format` method

### PR DESCRIPTION
**Pull request by lbajsarowicz**
### Description
As long as `format()` method lets you return formatted price with container - the return type is not float but string.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me {$VERSION} instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation

**Comment by magento-engcom-team** [comment]
@lbajsarowicz thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

File: lib/internal/Magento/Framework/Pricing/PriceCurrencyInterface.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function convertAndRound($amount, $scope = null, $currency = null, $preci
      * @param int $precision
      * @param null|string|bool|int|\Magento\Framework\App\ScopeInterface $scope
      * @param \Magento\Framework\Model\AbstractModel|string|null $currency
-     * @return float
+     * @return string
```
Comment (VladimirZaets) [inline lib/internal/Magento/Framework/Pricing/PriceCurrencyInterface.php:55]: Hi @lbajsarowicz, thanks for collaboration. Please declare strict return type according to [Magento technical guidelines](https://devdocs.magento.com/guides/v2.3/coding-standards/technical-guidelines.html)

**Review by VladimirZaets — CHANGES_REQUESTED** [review]

**Comment by lbajsarowicz** [comment]
@VladimirZaets I'm not sure if you expected so exceeded changes to this interface, but it's done.

Please be aware that there are 2 classes:
* `Magento\Directory\Model\PriceCurrency` (changed by me)
* `Magento\NegotiableQuote\Model\PriceCurrency` (Magento Commerce)

that implement this interface, so Commerce team has to change implementation of NegotiableQuote

Unit Tests failed on something that completely does not make sense for me:

https://github.com/magento/magento2/blob/a8d7f3715437a31880895bd7897b7557c2824203/app/code/Magento/Bundle/Test/Unit/Model/Product/TypeTest.php#L497-L499


In general - in cases just like this - it should not return object, but float even in unit tests.

Should I also fix the Unit Tests completely not related with my code, then?

**Comment by VladimirZaets** [comment]
@lbajsarowicz yes, you are right, our tests are also failed, so please revert your last changes. Thanks

**Comment by lbajsarowicz** [comment]
@VladimirZaets OK. Reverted latest changes. I believe this interface can be improved later on, but we need to improve unit tests first.

**Comment by VladimirZaets** [comment]
@lbajsarowicz The unit tests aren't a problem, there are simple to fix, the real problem is backward compatibility.

**Comment by VladimirZaets** [comment]
@lbajsarowicz Someone can using this interface but do not follow types that declared in the doc blocks and add strict types can break functionality. We should take into account this cases.

**Comment by lbajsarowicz** [comment]
Reverted the branch accidentally removed with cleanup.

I believe that we should follow the technical guidelines even with the price of backward compatibility for 2.3

`string` type is so generic, that changing any implementation should not be problematic.

**Pull request closed without merging**